### DISCUSSION TITLE
add_mesh_slice access to each slice of the mesh

### DISCUSSION THREAD

**Discussion by chautuankien** [pyvista/pyvista#4505] [Q&A]
I am using add_mesh_slice() to visualize the slice of my mesh data, and I was able to do that. Now, I want to create a slider which when I move the slider, the sliced mesh data will move to next slice. The max value of slider will be the number of slice in the mesh data. I want to view across x,y,z axis. Does anybody know how to do this?

**Comment by banesullivan** [comment]
This thread may have some code that will inspire a solution: https://github.com/pyvista/pyvista/discussions/1670

**Comment by edabor** [comment]
@chautuankien, Here is some code that might help you:

```python
import vtk
import numpy as np
import pyvista as pv
from pyvista import examples
from functools import partial

vol = examples.download_brain()

p = pv.Plotter()

def slice_mesh(value, normal=(1, 0, 0)):
    origin = np.asarray(normal) * value

    slice = vol.slice(normal=normal, origin=origin)
    p.add_mesh(slice, name=str(normal))
    return slice

slicer_x = partial(slice_mesh, normal=(1, 0, 0))
p.add_slider_widget(
    slicer_x,
    rng=vol.bounds[0:2],
    pointa=(0, 0.9),
    pointb=(0.2, 0.9),
    style="modern",
    title="x axis",
)

slicer_y = partial(slice_mesh, normal=(0, 1, 0))
p.add_slider_widget(
    slicer_y,
    rng=vol.bounds[2:4],
    pointa=(0, 0.7),
    pointb=(0.2, 0.7),
    style="modern",
    title="y axis",
)
slicer_z = partial(slice_mesh, normal=(0, 0, 1))
p.add_slider_widget(
    slicer_z,
    rng=vol.bounds[-2:],
    pointa=(0, 0.5),
    pointb=(0.2, 0.5),
    style="modern",
    title="z axis",
)
p.show()
```

But I guess that thanks to @banesullivan who added support for `vtkAlgoritms` in `Plotter.add_mesh`, (see https://docs.pyvista.org/version/stable/examples/99-advanced/plotting-algorithms.html), we could come up with a more elegant solution but I am not very familiar (yet) with this wonderful functionality.
